src/Common/Acl/AclMain.php

```diff
@@ -225,6 +225,32 @@ public static function aclCheckCore($section, $value, $user = '', $return_value
             }
         }
 
+        $remoteAddr = $_SERVER['REMOTE_ADDR'] ?? null;
+        $ips = explode("\r\n", $GLOBALS['white_list']);
+        $valid_IP = false;
+        for ($i = 0; $i < count($ips); $i++) {
+            $parts = explode("/", $ips[$i]);
+
+            // Checks if user IP is within the range of a stored IP with CIDR notation
+            if (count($parts) > 1) {
+                $bits_IP = ip2long($remoteAddr);
+                $bits_CIDR = ip2long($parts[0]);
+                $mask = str_repeat('1', intval($parts[1])) . str_repeat('0', 32 - intval($parts[1]));
+                if (($bits_IP & bindec($mask)) == ($bits_CIDR & bindec($mask))) {
+                    $valid_IP = true;
+                }
+            } else {
+                if ($remoteAddr == $ips[$i]) {
+                    $valid_IP = true;
+                }
+            }
+        }
+
+        // Allows access when white list of IPs is empty or if the user IP is in the white list
+        if (!empty($GLOBALS['white_list']) && !$valid_IP) {
+            exit();
```

Comment:
Instead of `exit()`'ing here you'd want to follow the convention of returning a `false` boolean to allow the upstream script to decide what to do next